aioaws/core.py

```diff
@@ -104,32 +105,39 @@ def add_signed_download_params(self, method: Literal['GET', 'POST'], url: URL, e
         assert expires >= 1, f'expires must be greater than or equal to 1, not {expires}'
         assert expires <= 604800, f'expires must be less than or equal to 604800, not {expires}'
         now = utcnow()
-        url = url.copy_merge_params(
-            {
-                'X-Amz-Algorithm': _AUTH_ALGORITHM,
-                'X-Amz-Credential': self._aws4_credential(now),
-                'X-Amz-Date': _aws4_x_amz_date(now),
-                'X-Amz-Expires': str(expires),
-                'X-Amz-SignedHeaders': 'host',
-            }
-        )
+        params = {
+            'X-Amz-Algorithm': _AUTH_ALGORITHM,
+            'X-Amz-Credential': self._aws4_credential(now),
+            'X-Amz-Date': _aws4_x_amz_date(now),
+            'X-Amz-Expires': str(expires),
+        }
+        if self.aws_session_token:
+            params.update({'X-Amz-Security-Token': self.aws_session_token})
+        params.update({'X-Amz-SignedHeaders': 'host'})
```

Comment:
```suggestion
            params['X-Amz-Security-Token'] = self.aws_session_token
        params['X-Amz-SignedHeaders'] = 'host'
```

Reply:
Is there any reason to put `host` after `X-Amz-Security-Token`?

Same question to the cases below.

Reply:
Yes, to keep the fields in alphabetical order. I get errors when the fields are out of order. You had a similar note for headers here:

https://github.com/samuelcolvin/aioaws/blob/9e938435da88bb0be26b21b834f37e6cd759e15e/aioaws/core.py#L146